### DISCUSSION TITLE
Is it hard to migrate and existing tkinter code to CustomTkinter ?

### DISCUSSION THREAD

**Discussion by Erim32** [TomSchimansky/CustomTkinter#1149] [Q&A]
I just discover CustomTkinter and I would know if this is possible to just replace tkinter Class with CustomTkinter class ( share same params) or if that require to rewrite all the code to migrate from one to the other library.

**Comment by Wolf-SO** [comment]
To me, porting seems quite easy, but I'm new to it too.

Apart from the additional instructions concerning style and theme, the port seems to be limited to prefixing the names of the classes with a `C`, e.g. using `CTk` instead of `Tk`, `CTkButton` instead of `TkButton`, etc. 

I think this is what [*Example Program*](https://github.com/TomSchimansky/CustomTkinter#example-program "TomSchimansky/CustomTkinter: A modern and customizable python UI-library based on Tkinter") suggests.

**Reply by VollGerne** [reply]
I have ported a small app from tkinter to customtkinter, was quite easy no problems in point of functionality.
BUT: I guess the reason we all move from tkinter to customtkinter is the very nice looking UI, so I had to get rid of the of the sizegrip as it doesn't fit in the customtkinter look (what is not a problem at all) and I will implement the tkinter.messagebox as new classes in a modal window in customtkinter, again, they don't fit anymore in the very nice looking GUI.

So from a functional point of view, it's easy, but if you want the same design throughout your application some extra effort is needed. But it's worth in my point of view
